colcon_defaults/argument_parser/defaults.py

```diff
@@ -112,6 +128,12 @@ def collect_parsers_by_verb(root, parsers, parent_verbs=()):
 
         data = self._get_defaults_values(self._config_path)
 
+        # Merge local defaults with global defaults, possibly overwriting
+        local_defaults_path = Path(WORKSPACE_DEFAULTS_FILE)
+        if local_defaults_path.is_file():
```

Comment:
`_get_defaults_values` is already performing this check, so we should skip it here.

Reply:
Good call, it's updated now.